### ISSUE TITLE
How to save Heatmap in loop

### ISSUE THREAD

**Issue by KoHaku55** [jokergoo/ComplexHeatmap#188]
Hi, I try to save heatmap in loop by using png() and dev.off() but it does not work. 
When I try to save it individually, it works. Do you know what is happen ? and How can I fix this ? 

Thank you.

**Comment by jokergoo** [comment]
You need to call `draw()` function explicitly:

```
for(...) {
    draw(Heatmap(...))
}
```

**Comment by Ahusnoo** [comment]
This works great!

**Comment by krushnach80** [comment]
> You need to call `draw()` function explicitly:
> 
> ```
> for(...) {
>     draw(Heatmap(...))
> }
> ```

My attempt to save multiple heatmaps with their respective filenames
`hm_lst<-cmplx_ht(list_of_files,metadata) ###  this generates my list of heatmaps

Now I want to save them into different files with their names


I tried this
```

lapply(1: length(hm_lst), function(i){
  ggsave(plot = hm_lst[[i]], 
         filename = paste(names(hm_lst)[i], ".pdf"), 
         width = 15)
})
```

The error I got is this
` Error: $ operator not defined for this S4 class 
`
`
Any suggestion how to do the same ? is ggsave the issue?

**Comment by krushnach80** [comment]
> I don't use `ggsave()`. Can you directly use `pdf()` to save the plots?

Yes it worked with pdf ..ggsave was not working ..thank you for your reply
